### PR TITLE
feat: added adr for node to rust foreign function interface

### PR DESCRIPTION
**Pull request by l0r1s**

### CODE REVIEW

**Comment by pepoviola** [comment]
Thanks @l0r1s, this looks great! I think we can use `wasm` for all config crates so we allow others to easily use the `sdk` and `napi` for orchestrator/providers, since those will require access to features not yet supported from wasm. 
I know that having this mix will include more work from our side to maintain but I also make sense to me using the best tool for each part of the `sdk`.

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+6. Copy artifacts :
+```
+mv napi-prototype.linux-x64-gnu.node index.d.ts index.js npm/linux-x64-gnu
+```
+
+7. Install package in ```ffi-prototype/app``` :
+```bash
+npm i ../napi-prototype/npm/linux-x64-gnu/
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:108]:
```suggestion
-> ffi-prototype/app:  npm i ../../napi-prototype/npm/linux-x64-gnu/
```

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+5. Build the project :
+```bash
+npm run build
+```
+
+6. Copy artifacts :
+```
+mv napi-prototype.linux-x64-gnu.node index.d.ts index.js npm/linux-x64-gnu
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:103]:
```suggestion
-> napi-prototype: mv napi-prototype.linux-x64-gnu.node index.d.ts index.js npm/linux-x64-gnu
```

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+## [Back](001-node-to-rust-foreign-function-interface.md)
+
+## Napi-rs prototype
+___
+
+1. Install the napi CLI
+
+```bash
+npm install -g @napi-rs/cli
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:9]:
```suggestion
-> napi-prototype: npm install -g @napi-rs/cli
```

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+2. Create a new napi project
+
+```bash
+napi new napi-prototype
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:15]:
```suggestion
-> napi-prototype: napi new napi-prototype
```

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+3. Install cargo dependencies
+
+```bash
+cargo add tokio --features full
+cargo add reqwest --features blocking
+cargo add napi --no-default-features --features napi4,async
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:23]:
```suggestion
-> napi-prototype: cargo add tokio --features full
-> napi-prototype: cargo add reqwest --features blocking
-> napi-prototype: cargo add napi --no-default-features --features napi4,async
```

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    thread_safe_callback.call(Ok(body.unwrap()), ThreadsafeFunctionCallMode::Blocking)
+  });
+
+  Ok(())
+}
+```
+
+5. Build the project :
+```bash
+npm run build
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:98]:
```suggestion
-> napi-prototype: npm run build
```

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  console.log(
+    `HTTP request through FFI with promise ${(await fetchPromise()).length}`
+  );
+})();
+```
+
+9. Build and execute the app :
+
+```bash
+npm run build+exec
```
Comment (wirednkod) [inline adrs/001-node-rust-foreign-function-interface/napi-prototype.md:130]:
```suggestion
-> ffi-prototype/app: npm run build+exec
```

**Review by wirednkod — CHANGES_REQUESTED** [review]

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md (hunk 3 above)
Comment (l0r1s) [reply adrs/001-node-rust-foreign-function-interface/napi-prototype.md]: Global package install so not needed

File: adrs/001-node-rust-foreign-function-interface/napi-prototype.md (hunk 4 above)
Comment (l0r1s) [reply adrs/001-node-rust-foreign-function-interface/napi-prototype.md:15]: We are not inside the directory, we are creating it with this command

**Comment by l0r1s** [comment]
Updated all markdown shell with the working directory @wirednkod

**Comment by l0r1s** [comment]
Set as accepted and updated decision outcome

**Review by pepoviola — APPROVED** [review]

**Pull request merged**